### PR TITLE
test: new integration tests

### PR DESCRIPTION
**Pull request by justasnortal**
Spring tests migrated to cucumber integration tests.
Integration test for calling all feign endpoints without authorization.
Bugfix to getVersion endpoint.

### CODE REVIEW

File: src/central-server/admin-service/int-test/src/intTest/java/org/niis/xroad/cs/test/glue/SecurityServerApiStepDefs.java [hunk 1]
```diff
@@ -198,4 +204,62 @@ private String randomSecurityServerId() {
...
+
+    private PagingSortingParametersDto getPagingSortingParameter(String sortBy, String order) {
+        final PagingSortingParametersDto dto = new PagingSortingParametersDto();
+        dto.setSort(sortBy);
+        dto.setDesc("desc".equalsIgnoreCase(order));
+        return dto;
+    }
+
+    @Step("the list is sorted by {string} {string}")
+    public void theListIsSortedByServerCodeDesc(String fieldExpression, String order) {
```
Comment (ricardas-buc) [inline src/central-server/admin-service/int-test/src/intTest/java/org/niis/xroad/cs/test/glue/SecurityServerApiStepDefs.java:227]: step sounds really generic, maybe it should be under separate definition file?

File: src/central-server/admin-service/int-test/src/intTest/resources/behavior/api/securityServerApi.feature [hunk 2]
```diff
@@ -10,6 +10,32 @@ Feature: Security Server API
...
+    And new security server 'CS:TEST:member-1:SS-X' authentication certificate registered
+    And management request is approved
+    And new security server 'CS:ANOTHER:member-2:SS-A' authentication certificate registered
+    And management request is approved
+    And Authentication header is set to SYSTEM_ADMINISTRATOR
+    When user requests security servers list sorted by '<$sortField>' '<$sortDirection>'
+    Then the list is sorted by '<$responseFieldExpression>' '<$sortDirection>'
+    Examples:
+      | $sortField            | $sortDirection | $responseFieldExpression          |
+      | xroad_id.server_code  | desc           | xroadId.serverCode.toUpperCase()  |
```
Comment (ricardas-buc) [inline src/central-server/admin-service/int-test/src/intTest/resources/behavior/api/securityServerApi.feature:31]: if we always uppercase it, maybe append it in stepDef?

File: src/central-server/admin-service/int-test/src/intTest/java/org/niis/xroad/cs/test/glue/MiscellaneousStepDefs.java [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+
+import static org.apache.commons.lang3.RandomStringUtils.randomAlphabetic;
+import static org.junit.Assert.fail;
+
+@SuppressWarnings("SpringJavaInjectionPointsAutowiringInspection")
+public class MiscellaneousStepDefs extends BaseStepDefs {
```
Comment (ricardas-buc) [inline src/central-server/admin-service/int-test/src/intTest/java/org/niis/xroad/cs/test/glue/MiscellaneousStepDefs.java:58]: maybe not misc steps, but ApiSecurityCheckStepDefs or similar?

**Review by ricardas-buc — CHANGES_REQUESTED** [review]

File: src/central-server/admin-service/int-test/src/intTest/java/org/niis/xroad/cs/test/glue/SecurityServerApiStepDefs.java (hunk 1 above)
Comment (justasnortal) [reply src/central-server/admin-service/int-test/src/intTest/java/org/niis/xroad/cs/test/glue/SecurityServerApiStepDefs.java]: Moved to CommonStepDefs

**Review by ricardas-buc — APPROVED** [review]

**Pull request merged**